### DISCUSSION TITLE
ILI9225 and ESP32

### DISCUSSION THREAD

**Discussion by ChassaB** [Bodmer/TFT_eSPI#2720] [Compatible displays and setup files]
Hi I have created a user setup for the ILI9225 (176*220) with ESP32. Works AOK with ESP DEV module on Arduino. Analogue clock example reworked for this screen size with a nice elliptical face. Don't know where I should put the files so others can use them if needed.

Guidance appreciated

**Comment by pierbout** [comment]
Hello. Where is that file? I don't see it in the user_setup directory.Thanks in advance.

**Reply by ChassaB** [reply]
Hi Pierre. Here are the settings for the file. 
 didn't get around to figuring out how to put it into the GIT directory


#define ILI9225_DRIVER //  2.2inch 176x220 display

// PIN numbers for my ESP32 Dev board
#define TFT_MISO 19 //Not used but defined. Do not use for anything else
#define TFT_MOSI 23
#define TFT_SCLK 18
#define TFT_CS   15  // Chip select control pin for a single screen.
#define TFT_DC   2  // Data Command control pin. Usually 2 but 2 is driving the onboard LED on my board
#define TFT_RST   4  // Reset pin

// Section 3 Fonts. All uncommented and available. Left font8 as normal

#define SPI_FREQUENCY  27000000 //didn't want to work with anything higher


#define SPI_READ_FREQUENCY  20000000


#define SPI_TOUCH_FREQUENCY  2500000

**Reply by sreyas40** [reply]
hello, im currently working with the same tft you mentioned. This is my user setup, but it is not working, just a white screen. I ran the read user_setup example, its showing the updated changes in the setup. To verify the pin, i used another library TFT_22_ILI225, and i was able to run the examples. Can you help me

`#define USER_SETUP_INFO "User_Setup"

#define ILI9225_DRIVER
#define TFT_RGB_ORDER TFT_RGB  // Colour order Red-Green-Blue

#define TFT_WIDTH  176
#define TFT_HEIGHT 220

//#define TFT_INVERSION_ON
 //#define TFT_INVERSION_OFF

//#define TFT_MISO 19
#define TFT_MOSI 13 // HSPI MOSI
#define TFT_SCLK 14  //HSPI CLK
#define TFT_CS   15  // HSPI CS
#define TFT_DC    27  // Data Command control pin
#define TFT_RST   33  // Reset pin 

#define LOAD_GLCD   
#define LOAD_FONT2  
#define LOAD_FONT4 
#define LOAD_FONT6  
#define LOAD_FONT7 
#define LOAD_FONT8  
//#define LOAD_FONT8N 
#define LOAD_GFXFF  
#define SMOOTH_FONT

#define SPI_FREQUENCY  27000000
#define SPI_READ_FREQUENCY  20000000
#define SPI_TOUCH_FREQUENCY  2500000
#define USE_HSPI_PORT`

**Reply by pierbout** [reply]
> hello, im currently working with the same tft you mentioned. This is my user setup, but it is not working, just a white screen. I ran the read user_setup example, its showing the updated changes in the setup. To verify the pin, i used another library TFT_22_ILI225, and i was able to run the examples. Can you help me



Hello Sreyas,

This is the User_Setup.h that I use, and it works fine. The Mosi, SCL, SDA, SCK and CS cables might have another number on your board. I also added my ILI9225 files (in TFT_Drivers directory). My 3.3 and led pins are directly connected to the 3.3v pin. Hope it will help you.

#define USER_SETUP_INFO "User_Setup"
#define ILI9225_DRIVER
#if defined(ESP32)
 #define USE_VSPI_PORT
  #define TFT_RST   4              // Reset pin (could connect to RST pin). Set it to -1 if the display RESET is connected to NodeMCU RST or 3.3V
  #define TFT_DC   2      //RS     // Data Command control pin
  #define TFT_CS   5      //SS     // Chip select control pin
  #define TFT_SDI  23     //MOSI
  #define TFT_MOSI 23
  #define TFT_CLK 18      //SCK
  #define TFT_SCLK 18
#endif

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts
#define SMOOTH_FONT
#define SPI_FREQUENCY  27000000
#define SPI_READ_FREQUENCY  20000000
#define SPI_TOUCH_FREQUENCY  2500000

[TFT_Drivers-ILI9225.zip](https://github.com/user-attachments/files/20643546/TFT_Drivers-ILI9225.zip)
![ESP32-D1-Board-liexpress-30_pins-PINS](https://github.com/user-attachments/assets/96a11adf-fe71-4dd4-a2d5-8ca16496d760)

**Reply by sreyas40** [reply]
Hello,
It didn't help. I tried using your pin config, as well as mine, still the blank white screen. When i replaced the driver files, it throw me an error in ILI9163_Rotation.h. In line 12, width  = initWidth; this was not declared, so i replaced it with _width  = _init_height; and the error was gone. Could you share your whole tft_eSPI library folder

**Comment by pierbout** [comment]
Thanks à lot!
